### PR TITLE
[Load] Support reading multi-line json objects for JsonScanner

### LINKED ISSUE #5773: [Load] Support reading multi-line json objects for JsonScanner
**Is your feature request related to a problem? Please describe.**
Currently, json load only supports importing one json object, rather than importing multiple lines of json objects, which is not conducive to improving the throughput of load. There are a large number of multi-line json files in the hdfs file, so we need to support reading multi-line json objects Scanner

**Describe the solution you'd like**
 support multi line json objects load

like the following：

{"name":"hello","id":5}
{"name":"2","id":1}
{"name":"2","id":1}
{"name":"cch","id":4}
{"name":"1000","id":9}
{"name":"4","id":3}
{"name":"cch","id":4}
{"name":"you","id":6}
{"name":"100","id":8}
{"name":"4","id":3}
{"name":"cch","id":4}
{"name":"2","id":1}
{"name":"4","id":3}
{"name":"hello","id":7}
{"name":"dahd","id":100}

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by caiconghui**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #5773 ) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/json_scanner.cpp [hunk 1]
```diff
@@ -141,13 +172,43 @@ Status JsonScanner::open_next_reader() {
...
+Status JsonScanner::open_line_reader() {
+    if (_cur_line_reader != nullptr) {
+        delete _cur_line_reader;
+        _cur_line_reader = nullptr;
+    }
+
+    const TBrokerRangeDesc& range = _ranges[_next_range];
+    int64_t size = range.size;
+    if (range.start_offset != 0) {
+        size += 1;
```
Comment (morningman) [inline be/src/exec/json_scanner.cpp:187]: Add comment to explain why +1 for size

File: be/src/exec/json_scanner.cpp [hunk 2]
```diff
@@ -270,27 +339,20 @@ void JsonReader::_close() {
...
+    // read a whole message
     SCOPED_TIMER(_file_read_timer);
-    std::unique_ptr<uint8_t[]> json_str;
-    size_t length = 0;
-    RETURN_IF_ERROR(_file_reader->read_one_message(&json_str, &length));
-    _bytes_read_counter += length;
-    if (length == 0) {
-        *eof = true;
+    const uint8_t* json_str = nullptr;
+    RETURN_IF_ERROR(_line_reader->read_line(&json_str, size, eof));
```
Comment (morningman) [inline be/src/exec/json_scanner.cpp:353]: What if the content of load file is a json object with multi line? like:
```
{
    "name" : "zs",
    "sex" : "male"
}
```
the first call of `read_line()` will return `"{"`?

Comment (caiconghui) [reply be/src/exec/json_scanner.cpp:353]: In fact, JSON usually serialized format does not contain line breaks. what you give above is JSON display format. If we encounter it, we will set the line_delimiter to other value

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**